### PR TITLE
Remove pytorch integration

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Motivation

Remove dependecies related to PyTorch.

## Description of the changes

Remove PyTorch Lightning, PyTorch Ignite, and PyTorch.
Please merge [this PR](https://github.com/optuna/optuna-integration/pull/62) first.

### CODE REVIEW

File: optuna/integration/pytorch_distributed.py [hunk 1]
```diff
@@ -1,353 +1,4 @@
...
-import pickle
-from typing import Any
-from typing import Callable
-from typing import Dict
-from typing import Optional
-from typing import overload
-from typing import Sequence
-from typing import TYPE_CHECKING
-from typing import TypeVar
+from optuna.integration.pytorch_distributed import TorchDistributedTrial
```
Comment (y0z) [inline optuna/integration/pytorch_distributed.py:1]:
```suggestion
from optuna_integration.pytorch_distributed import TorchDistributedTrial
```

File: optuna/integration/pytorch_ignite.py [hunk 2]
```diff
@@ -1,39 +1,4 @@
-import optuna
-from optuna.trial import Trial
+from optuna.integration.pytorch_ignite import PyTorchIgnitePruningHandler
```
Comment (y0z) [inline optuna/integration/pytorch_ignite.py:1]:
```suggestion
from optuna_integration.pytorch_ignite import PyTorchIgnitePruningHandler
```

File: optuna/integration/pytorch_lightning.py [hunk 3]
```diff
@@ -1,175 +1,4 @@
-import warnings
+from optuna.integration.pytorch_lightning import PyTorchLightningPruningCallback
```
Comment (y0z) [inline optuna/integration/pytorch_lightning.py:1]:
```suggestion
from optuna_integration.pytorch_lightning import PyTorchLightningPruningCallback
```

**Review by y0z — CHANGES_REQUESTED** [review]
I leave some change suggestions.

**Comment by y0z** [comment]
This PR progresses https://github.com/optuna/optuna/issues/4484.

**Comment by contramundum53** [comment]
@nabenabe0928 Could you move [these CI](https://github.com/optuna/optuna/blob/33b4eda9074ab6291612ed251e32acc0e0341f46/.github/workflows/tests-mpi.yml#L81-L84) to optuna_integration?

**Comment by nabenabe0928** [comment]
@contramundum53 Could you review this PR?

**Review by contramundum53 — APPROVED** [review]
LGTM!

**Review by y0z — APPROVED** [review]
LGTM

**Pull request merged**